framework/web/UploadedFile.php

```diff
@@ -149,25 +162,48 @@ public static function reset()
 
     /**
      * Saves the uploaded file.
-     * Note that this method uses php's move_uploaded_file() method. If the target file `$file`
-     * already exists, it will be overwritten.
-     * @param string $file the file path used to save the uploaded file
+     * If the target file `$file` already exists, it will be overwritten.
+     * @param string $file the file path or a path alias used to save the uploaded file.
      * @param bool $deleteTempFile whether to delete the temporary file after saving.
      * If true, you will not be able to save the uploaded file again in the current request.
      * @return bool true whether the file is saved successfully
      * @see error
      */
     public function saveAs($file, $deleteTempFile = true)
     {
-        if ($this->error == UPLOAD_ERR_OK) {
-            if ($deleteTempFile) {
-                return move_uploaded_file($this->tempName, $file);
-            } elseif (is_uploaded_file($this->tempName)) {
-                return copy($this->tempName, $file);
-            }
+        if ($this->error !== UPLOAD_ERR_OK) {
+            return false;
+        }
+        if (false === $this->copyTempFile(Yii::getAlias($file))) {
+            return false;
+        }
+        return !$deleteTempFile || $this->deleteTempFile();
+    }
+
+    /**
+     * @return bool|int
+     */
+    private function copyTempFile($file)
+    {
+        if (!is_resource($this->_tempResource)) {
+            return is_file($this->tempName) && copy($this->tempName, $file);
```

Comment:
That looks suspicious to me. Previously there was `is_uploaded_file` check that ensured that file moved is what came via POST. Now, resource is empty only when the method is POST but the check is missing.